src/Bicep.Core/Intermediate/ExpressionBuilder.cs

```diff
@@ -55,32 +56,36 @@ public Expression Convert(SyntaxBase syntax)
                     // Should have been caught earlier in validation.
                     _ => throw new NotImplementedException($"Unexpected out-of-range integer value"),
                 };
-                return new IntegerLiteralExpression(literalValue);
+                return new IntegerLiteralExpression(unary, literalValue);
             }
             case BooleanLiteralSyntax @bool:
-                return new BooleanLiteralExpression(@bool.Value);
+                return new BooleanLiteralExpression(@bool, @bool.Value);
             case NullLiteralSyntax:
-                return new NullLiteralExpression();
+                return new NullLiteralExpression(syntax);
             case ParenthesizedExpressionSyntax x:
                 return Convert(x.Expression);
```

Comment:
Should the parenthesized syntax be preserved as the source syntax for the underlying expression? E.g., `(foo.bar)` would convert to a `PropertyAccessExpression`, but should its source syntax be `foo.bar` or `(foo.bar)`?

Reply:
Personally, I find this implementation the most intuitive - the parentheses case is obviously a bit of a simple case, but as a general principle - assuming ARM gives us a reliable error position, it feels the most useful to map back to the smallest piece of Bicep syntax which generated the error (e.g. being as specific as we can be accurately).

It's also definitely an easier implementation, so I have a vested interest :)